components/common/src/ui.rs

```diff
@@ -432,46 +428,32 @@ impl UIReader for UI {
     }
 
     fn prompt_yes_no(&mut self, question: &str, default: Option<bool>) -> Result<bool> {
-        let ref mut stream = self.shell.out;
-        let choice = match default {
-            Some(yes) => {
-                if yes {
-                    match stream.is_colored() {
-                        true => format!(
-                            "{}{}{}",
-                            Colour::White.paint("["),
-                            Colour::White.bold().paint("Yes"),
-                            Colour::White.paint("/no/quit]")
-                        ),
-                        false => format!("[Yes/no/quit]"),
-                    }
-                } else {
-                    match stream.is_colored() {
-                        true => format!(
-                            "{}{}{}",
-                            Colour::White.paint("[yes/"),
-                            Colour::White.bold().paint("No"),
-                            Colour::White.paint("/quit]")
-                        ),
-                        false => format!("[yes/No/quit]"),
-                    }
-                }
+        let stream = &mut self.shell.out;
```

Comment:
The code below is a refactor. The existing version was awfully redundant and make the similarities hard to see across the colored/non-colored versions.